src/backends/googlepubsub/publisher.rs

```diff
@@ -0,0 +1,380 @@
+use crate::{Topic, ValidatedMessage};
+use futures_util::{
+    ready,
+    sink::{Sink, SinkExt},
+};
+use pin_project::pin_project;
+use std::{
+    collections::BTreeMap,
+    pin::Pin,
+    task::{Context, Poll},
+    time::SystemTime,
+};
+use ya_gcp::pubsub;
+
+use super::{
+    retry_policy, Connect, DefaultConnector, MakeConnection, PubSubError, SinkError, StatusCodeSet,
+    TopicName, Uri,
+};
+
+type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
```

Comment:
Import from the module above? Or, better, I feel define a newtype?

Reply:
it's actually only used for a type bound, not returned from functions or anything, so a newtype wouldn't help. I could probably just name the type in the bound instead of aliasing at all if that's more clear.